script.matchcenter/resources/lib/utilities/keymapeditor.py

```diff
@@ -0,0 +1,168 @@
+# -*- coding: utf-8 -*-
+'''
+    script.matchcenter - Football information for Kodi
+    A program addon that can be mapped to a key on your remote to display football information.
+    Livescores, Event details, Line-ups, League tables, next and previous matches by team. Follow what
+    others are saying about the match in twitter.
+    Copyright (C) 2016 enen92
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+'''
+
+import xbmc
+import xbmcgui
+import xbmcplugin
+import xbmcvfs
+import os
+import shutil
+import xml.etree.ElementTree as ET
+from xbmcgui import Dialog, WindowXMLDialog
+from threading import Timer
+from common_addon import *
+
+default = xbmc.translatePath('special://xbmc/system/keymaps/keyboard.xml')
+userdata = xbmc.translatePath('special://userdata/keymaps')
+gen_file = os.path.join(userdata, 'gen.xml')
+
+### Key mapp            
+def run():
+    ## load mappings ##
+    try:
+        setup_keymap_folder()
+    except Exception:
+        pass
+
+    defaultkeymap = read_keymap(default)
```

Comment:
````
except Exception:
    pass
````
I would never write such codes. this cant be accurately copied:) 

Seriously though, the problem with writing to the keymap file like this is that permanently overrides a key (including after uninstalling the addon) without users realizing it and with no obvious way to fix it. And they'll come to the forums as why key x doesn't work.

Why not just run the real keymap editor from here? It will be a few more steps for users but it'll at least be clearer what is going on and they'll have the option to go back to default.